CMakeLists.txt

```diff
@@ -64,6 +64,13 @@ if(USE_MKL_IF_AVAILABLE AND (NOT APPLE) AND (NOT MSVC) AND (CMAKE_HOST_SYSTEM_PR
 else()
   option(USE_MKLDNN "Build with MKL-DNN support" OFF)
 endif()
+#gcc 4 doesn't support AVX2 and SSSE3 support doesn't work with target attributes so ban gcc < 5 from intgemm.
+if ((CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64") AND (NOT CMAKE_CROSSCOMPILING) AND 
+    ((NOT CMAKE_COMPILER_IS_GNUCC) OR (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)))
+  option(USE_INTGEMM          "Build with x86 intgemm library for low-precision multiplication" ON)
+else()
+  option(USE_INTGEMM          "Build with x86 intgemm library for low-precision multiplication" OFF)
+endif()
```

Comment:
gcc7 is already required. You don't need any special handling here.

Reply:
d03342adbe0d68528b3e8e26c8517b4fbdb0eec9